core/plannerShared.h

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0
+#ifndef PLANNERSHARED_H
+#define PLANNERSHARED_H
+#include <QObject>
+
+// This is a shared class (mobile/desktop), and contains the core of the diveplanner
```

Comment:
Do I understand correctly, this is still the user facing part of the planner (i.e. the C++ part) not the internals (in C)?

Reply:
It it sitting in the middle, with the C code on one side and the UI/QML on the other.

E.g. UNIT_FACTOR, would need to be rebuild for qml and thus duplicated, this class avoids the duplication.

Reply:
Yea, this UNIT_FACTOR convertion thingie I didn't like at all. I know It's there since before, but I think we should clean that up.

We have a velocity_t in core/profile.h , but thats rather just colors for the profile. I think we should have a propper velocity_t which describes this and convertion helpers around that.

Reply:
I think mm/sec is the right unit for a velocity_t for the planner.